CRM/Contribute/BAO/Contribution.php

```diff
@@ -4234,21 +4234,18 @@ public static function completeOrder($input, $recurringContributionID, $contribu
       $contributionID = $contributionResult['id'];
     }
 
-    if ($input['component'] == 'contribute') {
-      if ($contributionParams['contribution_status_id'] === $completedContributionStatusID) {
-        self::updateMembershipBasedOnCompletionOfContribution(
-          $contributionID,
-          $input['trxn_date'] ?? date('YmdHis')
-        );
-      }
+    if ($contributionParams['contribution_status_id'] === $completedContributionStatusID) {
+      self::updateMembershipBasedOnCompletionOfContribution(
```

Comment:
@eileenmcnaughton if this is just for a participant payment will it just abort here? or is it the case that it will only be completed if its a recurring contribution which wouldn't apply to the participant payment?

Reply:
@seamuslee001 so the function it calls starts like this

```
    $memberships = self::getRelatedMemberships($contributionID);
    foreach ($memberships as $membership) {
```

ie any contribution without a membership (whether or not is has participant records) will have an empty array not do anything in that function